configapi/api_inventory.go

```diff
@@ -1,231 +1,231 @@
-    }
-    if newGnb.Tac == "" {
-        errorMessage := "post gNB request body is missing tac"
-        configLog.Errorf(errorMessage)
-        return errors.New(errorMessage)
-    }
-    req := httpwrapper.NewRequest(c.Request, newGnb)
-    procReq := req.Body.(configmodels.Gnb)
-    procReq.Name = gnbName
-    msg := configmodels.ConfigMessage{
-        MsgType:     configmodels.Inventory,
-        MsgMethod:   configmodels.Post_op,
-        GnbName: gnbName,
-        Gnb: &procReq,
-    }
-    configChannel <- &msg
-    configLog.Infof("successfully added gNB [%v] to config channel.", gnbName)
-    return nil
+	var gnbName string
+	var exists bool
+	if gnbName, exists = c.Params.Get("gnb-name"); !exists {
+		errorMessage := "post gNB request is missing gnb-name"
+		configLog.Errorf(errorMessage)
+		return errors.New(errorMessage)
+	}
+	configLog.Infof("received gNB %v", gnbName)
+	var err error
+	var newGnb configmodels.Gnb
+
+	allowHeader := strings.Split(c.GetHeader("Content-Type"), ";")
+	switch allowHeader[0] {
+	case "application/json":
+		err = c.ShouldBindJSON(&newGnb)
+	}
+	if err != nil {
+		configLog.Errorf("err %v", err)
+		return fmt.Errorf("failed to create gNB %v: %w", gnbName, err)
+	}
+	if newGnb.Tac == "" {
+		errorMessage := "post gNB request body is missing tac"
+		configLog.Errorf(errorMessage)
+		return errors.New(errorMessage)
+	}
+	req := httpwrapper.NewRequest(c.Request, newGnb)
+	procReq := req.Body.(configmodels.Gnb)
+	procReq.Name = gnbName
+	msg := configmodels.ConfigMessage{
+		MsgType:   configmodels.Inventory,
+		MsgMethod: configmodels.Post_op,
+		GnbName:   gnbName,
+		Gnb:       &procReq,
+	}
+	configChannel <- &msg
+	configLog.Infof("successfully added gNB [%v] to config channel.", gnbName)
+	return nil
 }
 
 func handleDeleteGnb(c *gin.Context) error {
-    var gnbName string
-    var exists bool
-    if gnbName, exists = c.Params.Get("gnb-name"); !exists {
-        errorMessage := "delete gNB request is missing gnb-name"
-        configLog.Errorf(errorMessage)
-        return errors.New(errorMessage)
-    }
-    configLog.Infof("received delete gNB %v request", gnbName)
-    msg := configmodels.ConfigMessage{
-        MsgType:   configmodels.Inventory,
-        MsgMethod: configmodels.Delete_op,
-        GnbName:   gnbName,
-    }
-    configChannel <- &msg
-    configLog.Infof("successfully added gNB [%v] with delete_op to config channel.", gnbName)
-    return nil
+	var gnbName string
+	var exists bool
+	if gnbName, exists = c.Params.Get("gnb-name"); !exists {
+		errorMessage := "delete gNB request is missing gnb-name"
+		configLog.Errorf(errorMessage)
+		return errors.New(errorMessage)
+	}
+	configLog.Infof("received delete gNB %v request", gnbName)
+	msg := configmodels.ConfigMessage{
+		MsgType:   configmodels.Inventory,
+		MsgMethod: configmodels.Delete_op,
+		GnbName:   gnbName,
+	}
+	configChannel <- &msg
+	configLog.Infof("successfully added gNB [%v] with delete_op to config channel.", gnbName)
```

Comment:
Do you mind making this update as part of your PR?
```suggestion
	configLog.Infof("successfully added gNB [%v] with delete_op to config channel", gnbName)
```